webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/VolumeTracingService.scala

```diff
@@ -149,6 +155,105 @@ class VolumeTracingService @Inject()(
     Fox.successful(tracing.withUserBoundingBoxes(updateUserBoundingBoxes()))
   }
 
+  def initializeWithDataMultiple(tracingId: String, tracing: VolumeTracing, initialData: File): Box[_] = {
+    if (tracing.version != 0L) {
+      return Failure("Tracing has already been edited.")
+    }
+    val mergedVolume = new MergedVolume
+
+    ZipIO.withUnziped(initialData) {
+      case (_, is) =>
+        val labelSet: mutable.Set[Byte] = scala.collection.mutable.Set()
+        ZipIO.withUnziped(is) {
+          case (_, is) =>
+            WKWFile.read(is) {
+              case (header, buckets) =>
+                if (header.numBlocksPerCube == 1) {
+                  val data: Array[Byte] = buckets.next()
+                  val nonZeroData: Array[Byte] = filterBytes(data, byte => byte != 0x00)
```

Comment:
In my opinion, the extraction isn't really necessary here, but it's not something that'll block this pr :slightly_smiling_face:

Reply:
I agree, it does not improve readability, in fact it may slightly degrade it. However, inlining this function breaks the code because of implicit conversions. This is why I wrote the `FilterBytes` trait, which does *not* extend `ProtoGeometryImplicits` and `KeyValueStoreImplicits`

Reply:
Ah, I see. Thanks for the explanation